src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -132,6 +136,13 @@
             CodeSegment,
             segment_kwargs={"type": "hash_prefix"},
         ),
+        RegexLexer(
+            "unquoted_relative_sql_file_path",
+            # currently there is no way to pass `regex.IGNORECASE` flag to `RegexLexer`
+            r"[.\w\\/#-]+\.[sS][qQ][lL]",
```

Comment:
Do we want this relative file path to be more general? I'm assuming (though can't verify), that you could use a file with other extensions here, eg. `./foo/bar.ddl` ?

Reply:
As far as I can tell, the [sqlcmd command `:r`](https://learn.microsoft.com/en-us/sql/tools/sqlcmd/sqlcmd-utility?view=sql-server-ver16#sqlcmd-commands:~:text=%3Ar%20%3C%20filename%20%3E%0AParses%20additional%20Transact%2DSQL%20statements%20and%20sqlcmd%20commands%20from%20the%20file%20specified%20by%20%3C%20filename%20%3E%20into%20the%20statement%20cache.) is explicitly for _"importing"_ the content of another `sql` file into another.

There are other statements in TSQL referencing files with different extensions (e.g. [OPENROWSET](https://learn.microsoft.com/en-us/sql/t-sql/functions/openrowset-transact-sql?view=sql-server-ver16#datasource)). However, these operate with single-quoted (and absolute) file paths.